cli/lotus/daemon.go

```diff
@@ -470,6 +470,25 @@ var DaemonCmd = &cli.Command{
 	},
 }
 
+// readGenesis detects if the path points to a zstd compressed file and if so
+// automatically decompress it. Otherwise, defaults to reading the path as is.
+func readGenesis(path string) ([]byte, error) {
+	genesisFile, err := os.Open(filepath.Clean(path))
+	if err != nil {
+		return nil, xerrors.Errorf("opening genesis file: %w", err)
+	}
+	defer func() { _ = genesisFile.Close() }()
+
+	switch compressed, err := build.IsZstdCompressed(genesisFile); {
+	case err != nil:
+		return nil, xerrors.Errorf("checking genesis header: %w", err)
+	case compressed:
+		return build.DecompressAsZstd(genesisFile)
+	default:
+		return io.ReadAll(genesisFile)
+	}
```

Comment:
personal preference, so take it or leave it, but I find there's a lower bounds on the complexity of `switch` statements where it's just easier to parse as a compound `if`, especially when this is the last thing in the function:

```suggestion
    if compressed, err := build.IsZstdCompressed(genesisFile); err != nil {
        return nil, xerrors.Errorf("checking genesis header: %w", err)
    } else if compressed {
        return build.DecompressAsZstd(genesisFile)
    }
    return io.ReadAll(genesisFile)
```

Reply:
Thank you for sharing. I find the switch statement specially s the last statement in a function with terminal `return`s far more readable and a good utilisation of golang syntax.

But using if and else doesn't bother me. Happy to change it if you find that easier to read.